app/src/main/java/com/amaze/filemanager/ui/fragments/FtpServerFragment.kt

```diff
@@ -407,13 +410,14 @@ class FtpServerFragment : Fragment(R.layout.fragment_ftp) {
                                 activityResultHandlerOnFtpServerPathGrantedSafAccess.launch(
                                     Intent(Intent.ACTION_OPEN_DOCUMENT_TREE).also {
                                         if (Build.VERSION.SDK_INT >= O &&
-                                            directoryUri.equals(DEFAULT_PATH)
+                                            directoryUri!!.startsWith(DEFAULT_PATH)
```

Comment:
No `!!` please

Reply:
I too know that would be silly... but with the default value given to `SharedPreferences.getString(key, defaultValue)` is a non-null, it's safe to assume `directoryUri` is definitely a non-null.

So there will still need a `!!` anyway. But I'm putting that up to the `SharedPreferences.getString()` call now.